### PR TITLE
Feature/107-110/endpoints for comments

### PR DESCRIPTION
**Pull request by OlkaA**
- Created migration for comments table.

- Created seeds for annotations and comments table.

- Issue #107 Created endpoint for creating comment.

- Issue#108 Created endpoint for listing comments for a specific annotation. 
Path for reading comments for a specific annotation looks like this:
comments/annotation/:annotation_id

- Issue#109 Created endpoint for updating a comment.

- Issue#110 Created endpoint for deleting a comment.

- Created endpoints for a swagger.

I had to comment lines with modules router(in router.js, api-router.js), because we don't have it right now.

### CODE REVIEW

File: src/server/api/routes/comments.router.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+// ENDPOINT: /api/comments/annotation/:fk_annotation_id :GET to get all comments added to a specific annotation
+router.get('/annotation/:fk_annotations_id', (req, res, next) => {
+  commentsController
+    .getCommentsForAnnotation(req.params.fk_annotations_id)
+    .then((result) => res.json(result))
+    .catch(next);
+});
+
+// ENDPOINT: /api/comments/ :PUT update comment for an annotation
```
Comment (laila1222) [inline src/server/api/routes/comments.router.js]: Really minor thing: in the commented section, you might have wanted to write:
// ENDPOINT:  /api/comments/**:comment_id**  :PUT update comment for an annotation

**Review by laila1222 — APPROVED** [review]
- Endpoints work 👍 
- Swagger documentation looks good 👍 
- Migrations and database with comments table loooks good 👍

File: src/server/api/routes/comments.router.js (hunk 1 above)
Comment (OlkaA) [reply src/server/api/routes/comments.router.js]: Thanks for noticing, resolved.

**Review by laila1222 — APPROVED** [review]
All routes and paths work! 👍 
Great job!

File: src/server/api/controllers/comments.controller.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  fk_annotations_id,
+  fk_comments_id,
+  fk_user_id,
+}) => {
+  await knex('comments').insert({
+    description: description,
+    fk_annotations_id: fk_annotations_id,
+    fk_comments_id: fk_comments_id,
+    fk_user_id: fk_user_id,
+  });
```
Comment (bytewiz) [inline src/server/api/controllers/comments.controller.js]: You could write this out like this:
```
await knex('comments').insert({
    description,
    fk_annotations_id,
    fk_comments_id,
    fk_user_id,
  });
```
Just fyi 🙌

File: src/server/api/controllers/comments.controller.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  return {
+    successful: true,
+  };
+};
+
+const getCommentsForAnnotation = (fk_annotations_id) => {
+  return knex('comments')
+    .where({ fk_annotations_id: fk_annotations_id })
```
Comment (bytewiz) [inline src/server/api/controllers/comments.controller.js]: same here 👍 `.where({ fk_annotations_id })`

File: src/server/api/controllers/comments.controller.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    .select('*');
+};
+
+const updateCommentById = (comment_id, body) => {
+  return knex('comments')
+    .where({ comment_id: comment_id })
```
Comment (bytewiz) [inline src/server/api/controllers/comments.controller.js]: and here `.where({ comment_id })`

File: src/server/api/controllers/comments.controller.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    .update({
+      description: body.description,
+      updated_at: new Date(),
+    });
+};
+
+const deleteCommentById = (comment_id) => {
+  return knex('comments')
+    .where({ comment_id: comment_id })
```
Comment (bytewiz) [inline src/server/api/controllers/comments.controller.js]: and here ```.where({ comment_id })```

**Review by bytewiz — APPROVED** [review]
Looks good 👍 Add some suggestion @OlkaA you can decide if you want to update or not 🙌

File: src/server/api/controllers/comments.controller.js (hunk 2 above)
Comment (OlkaA) [reply src/server/api/controllers/comments.controller.js]: Thanks for tips. 👍

**Pull request merged**